File: SantanderTeste/app/src/main/java/com/example/alessandrofsouza/santanderapp/data/service/ApiService.java
package com.example.alessandrofsouza.santanderapp.data.service;

import com.example.alessandrofsouza.santanderapp.domain.model.ContactModel;
import com.example.alessandrofsouza.santanderapp.domain.model.InvestmentModel;

import java.util.Set;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;
import retrofit2.http.GET;

public class ApiService {

    //private String BASE_URL = "https://floating-mountain-50292.herokuapp.com/";

    public ServiceApi getApi() {
        Retrofit retrofit = new Retrofit
                .Builder()
                .baseUrl(ServiceApi.BASE_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        return retrofit.create(ServiceApi.class);
    }

    public interface ServiceApi {

        public static final String BASE_URL = "https://floating-mountain-50292.herokuapp.com";

        @GET("/cells.json")
        Call<ContactModel> listCells();

        @GET("/fund.json")
        Call<InvestmentModel> listInvestment();

    }
}

File: app/src/main/java/com/fuinha11/test/testeandroid/service/api/ApiService.kt
package com.fuinha11.test.testeandroid.service.api

import com.fuinha11.test.testeandroid.BuildConfig
import com.fuinha11.test.testeandroid.service.api.data.response.CellsResponse
import com.fuinha11.test.testeandroid.service.api.data.response.FundResponse
import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import org.androidannotations.annotations.EBean
import retrofit2.Callback
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory


@EBean(scope = EBean.Scope.Singleton)
open class ApiService{

    protected val baseUrl = "https://floating-mountain-50292.herokuapp.com/"
    protected val api : ApiInterface

    init {
        val httpClient = OkHttpClient.Builder()

        if (BuildConfig.DEBUG) { //if in debug mode add logger to client
            val logger = HttpLoggingInterceptor()
            logger.level = HttpLoggingInterceptor.Level.BODY
            httpClient.addInterceptor(logger) //add logger to http client
        }

        val retrofit = Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .baseUrl(baseUrl)
                .client(httpClient.build())
                .build()
        api = retrofit.create(ApiInterface::class.java)
    }


    fun getCells(callback: Callback<CellsResponse>) {
        val call =  api.getCells()
        call.enqueue(callback)
    }

    fun getInvestment(callback: Callback<FundResponse>) {
        val call =  api.getInvestment()
        call.enqueue(callback)
    }
}